src/rust-crypto/rust-crypto.ts

```diff
@@ -1787,6 +1787,17 @@ class EventDecryptor {
             }
         }
 
+        // If we got a withheld code, expose that.
+        if (err.maybe_withheld) {
+            // Unfortunately the Rust SDK API doesn't let us distinguish between different withheld cases, other than
```

Comment:
Oh wow, this is because we are doing `to_string()` that uses `fmt::Display` instead of `as_str` that for ruma enum would give the code?

Reply:
So probably here https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/blob/a77375c0212a79a8650e81533c43801e559fff1e/src/error.rs#L57

Should be `as_str()` ?
I don't think it's intended that it's not the code here

Reply:
Yes, it's because the wasm bindings do this:

```
                maybe_withheld: withheld_code
                    .as_ref()
                    .map(|code| code.to_string().to_owned().into()),
```

Reply:
>  I don't think it's intended that it's not the code here

Guess there's a lesson in testing your code here!